app/src/main/java/com/gathering/android/event/home/FilterDialogFragment.kt

```diff
@@ -0,0 +1,50 @@
+package com.gathering.android.event.home
+
+import android.app.AlertDialog
+import android.app.Dialog
+import android.os.Bundle
+import android.view.KeyEvent
+import androidx.fragment.app.DialogFragment
+import com.gathering.android.R
+import com.gathering.android.databinding.FrgFilterDialogBinding
+
+class FilterDialogFragment(private val eventListViewModel: EventListViewModel) : DialogFragment() {
+
+    private lateinit var binding: FrgFilterDialogBinding
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+
+        binding = FrgFilterDialogBinding.inflate(layoutInflater)
+
+        val builder = AlertDialog.Builder(requireContext(), R.style.MyAlertDialogStyle)
+        builder.setView(binding.root)
+        builder.setTitle("Filter By")
+
+        builder.setNegativeButton("Cancel") { dialog, _ ->
+            dialog.dismiss()
+        }
+
+        val dialog = builder.create()
+        dialog.setCanceledOnTouchOutside(false)
+        dialog.setOnKeyListener { _, keyCode, _ ->
+            keyCode == KeyEvent.KEYCODE_BACK
+        }
```

Comment:
are you sure we need this ?

Reply:
By setting this OnKeyListener on the dialog, the code is essentially making the dialog dismiss when the user presses the back button. When the back button is pressed, the OnKeyListener will be triggered, and the code will check if the key pressed is the back button key. If it is, the dialog will be dismissed automatically

Reply:
`DialogFragment` does dismiss on its own when hitting back button. good call on removing this.